### PR TITLE
Simp reach estimates

### PR DESCRIPTION
**Pull request by alspellm**
All of the code used to produce simp reach estimates for 2016, 2019, 2021 added. 
Combined reach estimate and official plotting code also added.

### CODE REVIEW

File: processors/config/kalSimpTuple_cfg.py [hunk 1]
```diff
@@ -121,48 +139,57 @@
...
-cvtx.parameters["partCollRoot"]    = 'ParticlesOnCVertices_KF'
-cvtx.parameters["kinkRelCollLcio"] = ''
-cvtx.parameters["trkRelCollLcio"]  = 'KFTrackDataRelations'
-
-
 vtxgbl.parameters["debug"] = 0
 vtxgbl.parameters["vtxCollLcio"]     = 'UnconstrainedV0Vertices'
 vtxgbl.parameters["vtxCollRoot"]     = 'UnconstrainedV0Vertices'
-vtxgbl.parameters["partCollRoot"]    = 'ParticlesOnUVertices'
+vtxgbl.parameters["partCollRoot"]    = 'ParticlesOnVertices'
```
Comment (cbravo135) [inline processors/config/kalSimpTuple_cfg.py]: The unconstrained and constrained vtx should have different ROOT collection names.

File: processors/config/kalSimpTuple_cfg.py [hunk 2]
```diff
@@ -107,11 +119,17 @@
...
-ecal.parameters["hitCollLcio"] = 'EcalCalHits'
-ecal.parameters["hitCollRoot"] = ''#'RecoEcalHits'
-ecal.parameters["clusCollLcio"] = "EcalClustersCorr"
-ecal.parameters["clusCollRoot"] = "RecoEcalClusters"
+ecal_gbl.parameters["debug"] = 0
+ecal_gbl.parameters["hitCollLcio"] = 'EcalCalHits'
+ecal_gbl.parameters["hitCollRoot"] = 'RecoEcalHits'
+ecal_gbl.parameters["clusCollLcio"] = "EcalClustersCorr"
+ecal_gbl.parameters["clusCollRoot"] = "RecoEcalClusters"
+
```
Comment (cbravo135) [inline processors/config/kalSimpTuple_cfg.py]: What is the difference between ecal_gbl and ecal_kf? They look the same to me.

**Review by cbravo135 — COMMENTED** [review]
We might want to put all the simp selections in their own folder in analysis/selections and same for the plot configs in analysis/plotconfigs, these directories are starting to get a bit cluttered.

File: processors/config/kalSimpTuple_cfg.py (hunk 2 above)
Comment (alspellm) [reply processors/config/kalSimpTuple_cfg.py]: ah yeah, they were diff when I was running both recos at the same time. I will remove that now

File: processors/config/kalSimpTuple_cfg.py (hunk 1 above)
Comment (alspellm) [reply processors/config/kalSimpTuple_cfg.py]: whoops, Im not even sure why I removed that U, maybe thought it was a typo lol. Ill change it back

File: plotUtils/reach/simps22/2019_reach/hpstr_config/anaKalSimpTuple_cfg.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+elif (options.isData==0):
+    CalTimeOffset=43.
+    print("Running on MC file: Setting CalTimeOffset %d"  % CalTimeOffset)
+else:
+    print("Specify which type of ntuple you are running on: -t 1 [for Data] / -t 0 [for MC]")
+
+recoana_kf.parameters["CalTimeOffset"]=CalTimeOffset
+#Region definitions
+RegionPath=os.environ['HPSTR_BASE']+"/analysis/selections/"
```
Comment (cbravo135) [inline plotUtils/reach/simps22/2019_reach/hpstr_config/anaKalSimpTuple_cfg.py]: I think this needs to be updated

File: processors/config/kalSimpTuple_cfg.py [hunk 4]
```diff
@@ -121,48 +139,57 @@
...
-vtxgbl.parameters["partCollRoot"]    = 'ParticlesOnUVertices'
+vtxgbl.parameters["partCollRoot"]    = 'ParticlesOnVertices'
 vtxgbl.parameters["kinkRelCollLcio"] = 'GBLKinkDataRelations'
 vtxgbl.parameters["trkRelCollLcio"]  = 'TrackDataRelations'
 
 cvtxgbl.parameters["debug"] = 0
 cvtxgbl.parameters["vtxCollLcio"]     = 'TargetConstrainedV0Vertices'
 cvtxgbl.parameters["vtxCollRoot"]     = 'TargetConstrainedV0Vertices'
-cvtxgbl.parameters["partCollRoot"]    = 'ParticlesOnCVertices'
+cvtxgbl.parameters["partCollRoot"]    = 'ParticlesOnVertices'
```
Comment (cbravo135) [inline processors/config/kalSimpTuple_cfg.py]: There should be a "C" on this one, we could also using something more than just "C", but the important thing is it has a different name than the unconstrained vtx.

File: plotUtils/reach/simps22/2016_reach/hpstr_config/kalSimpTuple_cfg.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+vtx.parameters["vtxCollLcio"]    = 'UnconstrainedV0Vertices_KF'
+vtx.parameters["vtxCollRoot"]    = 'UnconstrainedV0Vertices_KF'
+vtx.parameters["partCollRoot"]   = 'ParticlesOnVertices_KF'
+vtx.parameters["kinkRelCollLcio"] = ''
+vtx.parameters["trkRelCollLcio"] = 'KFTrackDataRelations'
+
+vtxgbl.parameters["debug"] = 0
+vtxgbl.parameters["vtxCollLcio"]     = 'UnconstrainedV0Vertices'
+vtxgbl.parameters["vtxCollRoot"]     = 'UnconstrainedV0Vertices'
+vtxgbl.parameters["partCollRoot"]    = 'ParticlesOnVertices'
```
Comment (cbravo135) [inline plotUtils/reach/simps22/2016_reach/hpstr_config/kalSimpTuple_cfg.py]: This one doesn't have the fix to the name of the Particle/Vertex collections yet

File: processors/config/anaKalSimpTuple_cfg.py [hunk 6]
```diff
@@ -62,23 +66,42 @@
...
+    recoana_kf.parameters["regionDefinitions"] = [RegionPath+'Tight_2016_simp_reach_SR.json',
+                                                  RegionPath+'radMatchTight_2016_simp_reach_SR.json']
 
 #RecoHitAna
 recoana_gbl.parameters = recoana_kf.parameters.copy()
 recoana_gbl.parameters["anaName"] = "vtxana_gbl"
 recoana_gbl.parameters["vtxColl"] = "UnconstrainedV0Vertices"
 recoana_gbl.parameters["tsColl"]   = "TSData"
-recoana_gbl.parameters["hitColl"] = "RotatedHelicalOnTrackHits"
+recoana_gbl.parameters["hitColl"] = "RotatedHelicalTrackHits"
```
Comment (cbravo135) [inline processors/config/anaKalSimpTuple_cfg.py]: this should be "RotatedHelicalOnTrackHits"

File: processors/config/anaKalSimpTuple_cfg.py [hunk 7]
```diff
@@ -62,23 +66,42 @@
...
 recoana_gbl.parameters["trkColl"] = "GBLTracks"
-
+recoana_gbl.parameters["mcColl"]  = "MCParticle"
+recoana_gbl.parameters["ecalColl"] = "RecoEcalClusters"
+recoana_kf.parameters["vtxSelectionjson"] = os.environ['HPSTR_BASE']+"vertexSelection_2016_simp_reach.json"
+recoana_kf.parameters["histoCfg"] = os.environ['HPSTR_BASE']+"/analysis/plotconfigs/tracking/vtxAnalysis_2016_simp_reach.json"
```
Comment (cbravo135) [inline processors/config/anaKalSimpTuple_cfg.py]: These paths don't appear to be updated to me

**Review by cbravo135 — APPROVED** [review]

**Pull request merged**